docs/technical/git-submodules.md

```diff
@@ -65,6 +65,73 @@ pointing to `<superproject-git-directory>/modules/<submodule-name>`. The latter
 is sometimes called the "absorbed form", and is Git's preferred mode of
 operation.
 
+## Roadmap
+
+Git submodules should be implemented in an order that supports an increasing set
+of workflows, with the goal of getting feedback early and often. When support is
+incomplete, jj should not crash, but instead provide fallback behavior and warn
+the user where needed.
+
+The goal is to land good support for pure Jujutsu repositories, while colocated
+repositories will be supported when convenient.
+
+This section should be treated as a set of guidelines, not a strict order of
+work.
+
+### Phase 1: Readonly submodules
+
+This includes work that inspects submodule contents but does not create new
+objects in the submodule. This requires a way to store submodules in a jj
+repository that supports readonly operations.
+
+#### Outcomes
+
+- Submodules can be cloned anew
+- New submodule commits can be fetched
+- Submodule history and branches can be viewed
+- Submodule contents are populated in the working copy
+- Superproject gitlink can be updated to an existing submodule commit
+- Conflicts in the superproject gitlink can be resolved to an existing submodule
+  commit
```

Comment:
This doesn't mention when the submodules are cloned or when commits are fetched. Is that intentionally left unspecified for now (fine with me)?

Reply:
> This doesn't mention when the submodules are cloned or when commits are fetched. Is that intentionally left unspecified for now (fine with me)?

Yes, it is intentionally underspecified. I have some thoughts on how we might do this more ergonomically than Git, but they deserve their own discussion, so I'll send those thoughts as a separate PR.